diff --git a/cmake/FindCMath.cmake b/cmake/FindCMath.cmake
index dd42aba..930131c 100644
--- a/cmake/FindCMath.cmake
+++ b/cmake/FindCMath.cmake
@@ -31,7 +31,11 @@ include(CheckSymbolExists)
 include(CheckLibraryExists)
 
 check_symbol_exists(pow "math.h" CMath_HAVE_LIBC_POW)
-find_library(CMath_LIBRARY NAMES m PATHS ${CMAKE_C_IMPLICIT_LINK_DIRECTORIES})
+if(${VCPKG_TARGET_TRIPLET} STREQUAL "arm64-linux-static")
+    find_library(CMath_LIBRARY NAMES libm.a PATHS ${CMAKE_C_IMPLICIT_LINK_DIRECTORIES})
+else()
+    find_library(CMath_LIBRARY NAMES m PATHS ${CMAKE_C_IMPLICIT_LINK_DIRECTORIES})
+endif()
 
 if(NOT CMath_HAVE_LIBC_POW)
     set(CMAKE_REQUIRED_LIBRARIES_SAVE ${CMAKE_REQUIRED_LIBRARIES})
